pilot/pkg/serviceregistry/kube/controller/controller.go

```diff
@@ -180,8 +180,11 @@ func NewController(client kubernetes.Interface, options Options) *Controller {
 	svcInformer := sharedInformers.Core().V1().Services().Informer()
 	out.services = out.createCacheHandler(svcInformer, "Services")
 
-	epInformer := sharedInformers.Core().V1().Endpoints().Informer()
-	out.endpoints = out.createEDSCacheHandler(epInformer, "Endpoints")
+	if features.EnableEndpointSliceController {
```

Comment:
Instead of using the environment variable here, could we just pass it in as an option? The environment variable would be better handled as part of application bootstrap logic.  This would also make testing both code paths a bit easier (no mucking about with changing the env).

Reply:
done